tiledb/cloud/taskgraphs/client_executor/_replacers.py

```diff
@@ -81,3 +84,71 @@ def maybe_replace(self, arg) -> Optional[visitor.Replacement]:
         # If it was neither an escape sequence or a node output,
         # do not dig further into the value.
         return visitor.Replacement(arg)
+
+
+def visit_args(
+    vtor: visitor.ReplacingVisitor, args: types.Arguments
+) -> types.Arguments:
+    return types.Arguments(
+        (vtor.visit(a) for a in args.args),
+        {k: vtor.visit(v) for k, v in args.kwargs.items()},
+    )
+
+
+UDFArgument = Dict[str, Any]
+"""Marker for a structured dict for an argument, with ``name`` and ``value``."""
+
+
+def parse_json_args(json_args: List[UDFArgument]) -> types.Arguments:
+    """Parses a list of ``UDFArgument``s into an ``Arguments`` instance.
+
+    This takes the ``UDFArgument`` sequence and turns it into its ``args`` and
+    ``kwargs`` values, without performing any parsing of the actual values
+    inside each of the arguments (that is, it turns the JSON list into a
+    list of args and a dict of kwargs, but does not perform further processing).
+    """
+    # Usually we don't include internal type-checks, but this function is also
+    # used on the server-side UDF executor, where we do want to ensure that
+    # we don't get the wrong type of data.
+    arg_type = type(json_args)
+    if arg_type is not list:
+        raise ValueError(
```

Comment:
Good question! This is data that comes out of `json.Parse`, so any (non-`str`) sequences that we get will always be `list`s rather than anything else. This is technically not *strictly* necessary but is a useful check to ensure that everything is parsing as we expect.